Documentation/dotnet-primer.md

```diff
@@ -0,0 +1,147 @@
+.NET Primer
+===========
+
+.NET is a general purpose development platform. It can be used for any kind of app type or workload where general purpose solutions are used. It has several key features that are attractive to many developers, including automatic memory management and modern programming languages, that make it easier to efficiently build high-quality apps. .NET enables a high-level programming environment with many convenience features, while providing low-level access to native memory and APIs. 
+
+Multiple implementations of .NET are available, based on open [.NET Standards](dotnet-standards.md) that specify the fundamentals of the platform. They are separately optimized for different app types (e.g. desktop, mobile, gaming, cloud) and support many chips (e.g. x86/x64, ARM) and operating systems (e.g. Windows, Linux, iOS, Android, OS X). Open source is also an important part of the .NET ecosystem, with multiple .NET implementations and many libraries available under OSI-approved licenses.
+
+Defining .NET Features
+======================
+
+There are a set of key features that together define .NET. Most of them are not unique on their own, but the peculiar aggregation of these features is what defines .NET as being distinct.
+
+- [Automatic memory management](garbage-collection.md)
+- Verifiable code - type and memory safety
+- Flexible native code compilation (enables JIT and AOT)
+- Fast native interop and access to native memory
+- Operating system agnostic 
+- Chip-agnostic (e.g. x86, ARM64) byte-code
+- Language-agnostic (e.g. C#, F#, VB) runtime
+- Flexible type system
+- Expressive (imperative and functional) programming languages
+- Programmable source code compiler (Roslyn)
+- Large standard class library
+- [.NET Class Libraries](class-libraries.md)
+- [Assembly format](assembly-format.md)
+- Package management
+
+TODO: Ensure that this list is correct. Write and link to a chapter for each of these defining features.
+
+.NET by Example
+===============
+
+.NET includes many features and capabilities that you will use on a daily basis for common tasks and needs. You can see these described below, with examples. Some of the same topics are discussed in more detail in documents linked above.
+
+Automatic Memory Management
+---------------------------
+
+[Garbage collection](garbage-collection.md) is the most well-known of .NET features. Developers do not need to actively manage memory, athough there are affordances to provide more information to the garbage collector (GC). C# includes the `new` keyword to allocate memory in terms of a particular type, and the `using` keyword to scope the usage of an object. The GC operates on a lazy approach to memory management, preferring application throughput to the immediate collection of memory.
```

Comment:
`using` only works with `IDisposable` objects, I'm not sure introducing it at this stage is a good idea.